src/domains/transaction/hooks/use-pending-transactions.ts

```diff
@@ -42,6 +54,43 @@ export const usePendingTransactions = (): UsePendingTransactionsReturn => {
 		[setPendingTransactions],
 	);
 
+	const addPendingTransactionFromUnconfirmed = useCallback(
+		(input: UnconfirmedTransaction & { walletAddress: string; networkId: string; gasLimit?: string | number }) => {
+			try {
+				const gasLimitLike = (input as any).gasLimit ?? input.gas;
+				const signedData: RawTransactionData = {
+					data: input.data,
+					from: input.from,
+					gasLimit: Number(gasLimitLike ?? 0),
+					gasPrice: input.gasPrice as any,
+					hash: input.hash,
+					nonce: BigNumber.make(input.nonce),
+					senderPublicKey: (input as any).senderPublicKey,
+					to: input.to,
+					value: input.value,
+				};
+
+				const pending: PendingTransactionData = {
+					networkId: input.networkId,
+					transaction: {
+						serialized: "",
+						signedData,
+					},
+					walletAddress: input.walletAddress,
```

Comment:
here similar, the data already exists on the `input` and it feels obsolete to add it again as a meta field. If an address is necessary to determine the address it belongs to, I'd rather have it be stored by address in localstorage so we already filter on a higher level